### PR TITLE
add func to exclude launcher db from time machine

### PR DESCRIPTION
**Pull request by James-Pickett**

### CODE REVIEW

File: cmd/launcher/internal/timemachine/timemachine_darwin.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
+//go:build darwin
+// +build darwin
+
+package timemachine
```
Comment (James-Pickett) [inline cmd/launcher/internal/timemachine/timemachine_darwin.go:4]: had no idea where to put this 🤷

Comment (RebeccaMahany) [reply cmd/launcher/internal/timemachine/timemachine_darwin.go:4]: Could see it living in `ee/agent/`, maybe?

Comment (James-Pickett) [reply cmd/launcher/internal/timemachine/timemachine_darwin.go:4]: yea, that seems better

File: ee/agent/timemachine/timemachine_darwin.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+)
+
+// ExcludeLauncherDB adds the launcher db to the time machine exclusions for
+// darwin and is noop for other oses
+func ExcludeLauncherDB(ctx context.Context, k types.Knapsack) {
+	dbPath := k.BboltDB().Path()
+	cmd, err := allowedcmd.Tmutil(ctx, "addexclusion", dbPath)
+	if err != nil {
+		k.Slogger().Log(ctx, slog.LevelError,
+			"failed to add launcher db to time machine exclusions",
```
Comment (RebeccaMahany) [inline ee/agent/timemachine/timemachine_darwin.go:21]: Usually this error would be tmutil not being found --

```suggestion
			"could not create tmutil command to add launcher db to time machine exclusions",
```

**Review by RebeccaMahany — APPROVED** [review]

File: ee/agent/timemachine/timemachine_darwin.go (hunk 2 above)
Comment (James-Pickett) [reply ee/agent/timemachine/timemachine_darwin.go]: updated

**Review by RebeccaMahany — APPROVED** [review]
nice

File: ee/agent/timemachine/timemachine_darwin.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+	"github.com/kolide/launcher/ee/agent/types"
+	"github.com/kolide/launcher/ee/allowedcmd"
+)
+
+// ExcludeLauncherDB adds the launcher db to the time machine exclusions for
+// darwin and is noop for other oses
+func ExcludeLauncherDB(ctx context.Context, k types.Knapsack) {
+	dbPath := k.BboltDB().Path()
+	cmd, err := allowedcmd.Tmutil(ctx, "addexclusion", dbPath)
```
Comment (directionless) [inline ee/agent/timemachine/timemachine_darwin.go:18]: I think we should exclude the whole root directory. Maybe the secret too?

**Review by directionless — COMMENTED** [review]
I love how simple this is

File: ee/agent/timemachine/timemachine_darwin.go (hunk 3 above)
Comment (directionless) [reply ee/agent/timemachine/timemachine_darwin.go:18]: Actually keep the secret. But probably ignore the whole root directory

Comment (James-Pickett) [reply ee/agent/timemachine/timemachine_darwin.go:18]: I thought about that, but if I understand correctly, that would mean we would also not bring along all the updates. So after a time machine restore / backup the user would be back on the very first version of launcher they downloaded and would need to update again. Maybe that's okay?

Comment (directionless) [reply ee/agent/timemachine/timemachine_darwin.go:18]: Grabbing the updates, and the TUF db, is a great counterpoint. I think you're right, it's good to carry those. 

Maybe we should drop some more stuff though. Can we go by wildcard? Or do we need to do own globbing?

Comment (James-Pickett) [reply ee/agent/timemachine/timemachine_darwin.go:18]: we can wild card, you thinking `*.db`?

Comment (directionless) [reply ee/agent/timemachine/timemachine_darwin.go:18]:
```
$ ls /var/kolide-k2/k2device-preprod.kolide.com/
augeas-lenses				menu.json
debug-2023-12-31T20-27-14.599.json.gz	menu_template.json
debug-2024-01-03T05-30-26.256.json.gz	metadata.json
debug-2024-01-06T19-20-54.029.json.gz	metadata.plist
debug-2024-01-09T01-44-56.485.json.gz	osquery.autoload
debug-2024-01-10T20-30-07.640.json.gz	osquery.db
debug.json				osquery.pid
desktop_501				osquery.sock
kolide.png				osquery.sock.34719
kv.sqlite				osquery.sock.3513
launcher-staging			osqueryd-staging
launcher-tuf				osqueryd-tuf
launcher-tuf-dev			osqueryd-tuf-dev
launcher-version-1.4.1-3-ge21cebc	tuf
launcher.db				tuf-dev
launcher.pid				updates
```

json, json.gz, the metadata, the menu, most of the osquery files, desktop, the pid files, sockets...

Comment (James-Pickett) [reply ee/agent/timemachine/timemachine_darwin.go]: wildcards did not work as expected, ended up globbing

File: ee/agent/timemachine/timemachine_darwin.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+		"augeas-lenses",
+		"*.plist",
+		"osquery*",
+	}
+
+	var exclusionPaths []string
+	for _, pattern := range exclusionPatterns {
+		matches, err := filepath.Glob(filepath.Join(k.RootDirectory(), pattern))
+
+		if err != nil {
```
Comment (directionless) [inline ee/agent/timemachine/timemachine_darwin.go:52]:
```suggestion
		if err != nil {
```

**Review by directionless — APPROVED** [review]

**Review by RebeccaMahany — APPROVED** [review]

**Pull request merged**